File: execution/check_supabase.py
# execution/check_supabase.py
import os
from dotenv import load_dotenv
from supabase import create_client

load_dotenv()

SUPABASE_URL = os.getenv("NEXT_PUBLIC_SUPABASE_URL")
SUPABASE_KEY = os.getenv("NEXT_PUBLIC_SUPABASE_ANON_KEY")

supabase = create_client(SUPABASE_URL, SUPABASE_KEY)

try:
    print("Testing table query...")
    res = supabase.table("market_data").select("time, vix, esf").limit(1).execute()
    print("Basic columns work:", res.data)
except Exception as e:
    print("Basic columns error:", e)

try:
    print("Testing 'spx' column query...")
    res_spx = supabase.table("market_data").select("spx").limit(1).execute()
    print("spx column exists! Data:", res_spx.data)
except Exception as e:
    print("spx column does NOT exist or has error:", e)

File: frontend/check_supabase.js
// frontend/check_supabase.js
const { createClient } = require('@supabase/supabase-js');
require('dotenv').config({ path: '.env.local' });

const supabaseUrl = process.env.NEXT_PUBLIC_SUPABASE_URL;
const supabaseKey = process.env.NEXT_PUBLIC_SUPABASE_ANON_KEY;

const supabase = createClient(supabaseUrl, supabaseKey);

async function check() {
  try {
    const { data, error } = await supabase
      .from('market_data')
      .select('time, vix, esf')
      .limit(1);
    console.log('Columns time, vix, esf:', { data, error });

    const { data: dataSpx, error: errorSpx } = await supabase
      .from('market_data')
      .select('spx')
      .limit(1);
    console.log('Column spx:', { dataSpx, error: errorSpx?.message || errorSpx });
  } catch (e) {
    console.error('Err:', e);
  }
}

check();
